src/v/cluster/feature_frontend.cc

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "feature_frontend.h"
+
+#include "cluster/cluster_utils.h"
+#include "cluster/feature_table.h"
+#include "cluster/health_monitor_frontend.h"
+#include "cluster/members_table.h"
+
+namespace cluster {
+
+static constexpr std::chrono::seconds status_retry = 5s;
+
+ss::future<> feature_frontend::start() {
+    vlog(clusterlog.info, "Starting...");
+    _leader_notify_handle
+      = _group_manager.local().register_leadership_notification(
+        [this](
+          raft::group_id group,
+          model::term_id term,
+          std::optional<model::node_id> leader_id) {
+            vlog(
```

Comment:
this will be really verbose since we are logging before checking if notification is coming from `raft0`